packages/next-common/components/fellowship/core/members/actions/more/submitEvidenceItem/popup.jsx

```diff
@@ -0,0 +1,151 @@
+import TxSubmissionButton from "next-common/components/common/tx/txSubmissionButton";
+import useSigner from "next-common/components/common/tx/useSigner";
+import Editor from "next-common/components/editor";
+import PopupLabel from "next-common/components/popup/label";
+import PopupWithSigner from "next-common/components/popupWithSigner";
+import { usePopupParams } from "next-common/components/popupWithSigner/context";
+import { useContextApi } from "next-common/context/api";
+import { useUploadToIpfs } from "next-common/hooks/useUploadToIpfs";
+import { incFellowshipCoreMembersTrigger } from "next-common/store/reducers/fellowship/core";
+import { newErrorToast } from "next-common/store/reducers/toastSlice";
+import { cn } from "next-common/utils";
+import useRealAddress from "next-common/utils/hooks/useRealAddress";
+import useWaitSyncBlock from "next-common/utils/hooks/useWaitSyncBlock";
+import { useCallback, useState } from "react";
+import { useDispatch } from "react-redux";
+
+function WishChoice({ title, description, checked, onClick = noop }) {
+  return (
+    <div
+      className={cn(
+        "flex grow py-[10px] pl-[16px] pr-[10px] rounded-[8px] border border-neutral400 gap-[10px]",
+        "cursor-pointer",
+      )}
+      onClick={onClick}
+    >
+      <div className="flex flex-col grow">
+        <span className="text14Medium text-textPrimary">{title}</span>
+        <span className="text12Medium text-textTertiary">{description}</span>
+      </div>
+      <div>
+        {checked ? (
+          <SystemRadioButtonOn className="[&_path]:fill-theme500" />
+        ) : (
+          <SystemRadioButtonOff />
+        )}
+      </div>
+    </div>
+  );
+}
+
+function Content() {
+  const { onClose } = usePopupParams();
+  const dispatch = useDispatch();
+  const address = useRealAddress();
+  const { component } = useSigner("Address");
+  const [evidence, setEvidence] = useState("");
+  const [wish, setWish] = useState("retention");
+  const { uploading, upload } = useUploadToIpfs();
+
+  const api = useContextApi();
+
+  const getTxFunc = useCallback(async () => {
+    const { error, result } = await upload(
+      new File([evidence], `evidence-${address}-${wish}.txt`, {
+        type: "text/plain",
+      }),
+    );
+    if (error) {
+      dispatch(newErrorToast("Failed to upload evidence to IPFS"));
+      return;
+    }
+    const { cid } = result;
+
+    let digest;
+    try {
+      digest = CID.parse(cid).toV0().multihash.digest;
+    } catch (e) {
+      dispatch(newErrorToast("Failed to parse CID digest"));
+      return;
+    }
+
+    const hexDigest = "0x" + Buffer.from(digest).toString("hex");
+    return api.tx.fellowshipCore?.submitEvidence(wish, hexDigest);
+  }, [api, address, upload, evidence, wish, dispatch]);
+
+  const fnWaitSync = useWaitSyncBlock("Evidence submitted", () => {
+    dispatch(incFellowshipCoreMembersTrigger());
+  });
+  const onSubmit = (_, blockHash) => blockHash && fnWaitSync(blockHash);
```

Comment:
Why do we need update core members after evidence submission?

Reply:
refactor to use subscribe data instead